src/scripts/components/layer/wv.layer.list.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import LayerRadio from './wv.layer.radio.js';
+import { CellMeasurer, CellMeasurerCache, List } from 'react-virtualized';
+
+/*
+ * A react component, Builds a list of layers using the LayerRadio component
+ * @class LayerList
+ * @extends React.Component
+ */
+export default class LayerList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      layerFilter: props.config.layerOrder,
+      expandedLayers: []
+    };
+    this._cache = new CellMeasurerCache({
+      fixedWidth: true,
+      minHeight: 23,
+      minWidth: props.width
+    });
+  }
+
+  /*
+   * Clear cell catch if component will
+   *  rerender
+   * @method componentWillUpdate
+   * @return {void}
+   */
+  componentWillUpdate(){
+    this._cache.clearAll();
+  }
+
+  /*
+   * Saves the visibility/state of the metadata for each layer
+   * @method saveExpandedInfoState
+   * @param {string} layerID - the layer which has metadata that
+   *  needs to be toggled
+   * @return {void}
+   */
+  saveExpandedInfoState(layer){
+    var { expandedLayers } = this.state;
+    var index = expandedLayers.indexOf(layer);
+    if(index > -1){
+      expandedLayers.splice(index, 1); // Removes layer from expanded list
+    } else {
+      expandedLayers.push(layer);
+    }
+    this.setState({ expandedLayers: expandedLayers });
+  }
+
+  /*
+   * Recalculates the row height for a given rowIndex
+   * @method recalculateRowHeight
+   * @param {number} index - Index of the row to be recalculated
+   * @return {void}
+   */
+  recalculateRowHeight (rowIndex){
+    this._cache.clear(rowIndex, 0);
+    this._layerList.recomputeRowHeights(rowIndex);
+    this._layerList.scrollToRow(rowIndex);
+  }
+
+  /*
+   * A sub-rendering component
```

Comment:
What do you think about making this description say "Renders a single row in the layer search results list" or something more descriptive?

Reply:
Or the description from `react-virtualized` docs; "Responsible for rendering a single row, given its index."

Reply:
Even better, we should just link to the `react-virtualized` docs: https://github.com/bvaughn/react-virtualized/blob/master/docs/List.md#rowrenderer